### ISSUE TITLE
The Remote Scene Tree causes jerky movement everytime it updates

### ISSUE THREAD

**Issue by mrcdk** [godotengine/godot#13219]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Godot 3.0 master https://github.com/godotengine/godot/commit/3a33725014169ce51d6ac581a4dce3b97a60da00

**Issue description:**

In the example, Every time the Remote Scene Tree updates it causes some choppy movement. If I disable the auto switching to it under Editor settings / Debugger the issue disappear.

**Steps to reproduce:**
1. Open the example and make sure the remote scene tree is opened.

**Link to minimal example project:**

[level_test.zip](https://github.com/godotengine/godot/files/1500106/level_test.zip)

**Comment by kubecz3k** [comment]
As far as I'm able to tell I was able to observe the effect. Not sure if there is much we can do about it?
cc @Geequlim

**Comment by kubecz3k** [comment]
Also for now it might worth to consider if `Auto switch to remote scene tree` should be set on by default.

**Comment by Zylann** [comment]
Also, this reminds me this https://github.com/godotengine/godot/issues/8165

**Comment by kubecz3k** [comment]
@Zylann huh seems like a duplicate... with the exception that as far as I can tell currently its enough to hide remote tree (open local instead of remote)

**Comment by reduz** [comment]
Should be modified to send this information over many frames to avoid stall

**Comment by Zylann** [comment]
@kubecz3k folded nodes should be taken in consideration though

**Comment by Geequlim** [comment]
I observe of your example project. The choppy movement also exist  on my laptop but it doesn't disappear after I switched to local scene tree and inspect local object.

It didn't slow down or have anything changed even when I update the remote scene tree and remote inspector more frequently by changing `Remote Scene Tree Refresh Interval` and `Remote Inspector Referesh Interval` to `0.1`.

My Hardware and driver information:

Arch Linux with Mesa DRI Intel(R) HD Graphics 520 (Skylake GT2)

**Comment by AWKgamedev** [comment]
There's substantial stuttering for me, the graph is for a scene with about 2500 nodes.
![godot engine - harmon_i_c - level 1-2 tscn](https://user-images.githubusercontent.com/10258315/33482122-b1b547c2-d67e-11e7-9c76-158fa50fe60f.png)

**Comment by akien-mga** [comment]
Until the performance is improved, I suggest to make the remote scene tree opt-in. You don't need it all the time, only when you need to debug the scene tree.

**Comment by Geequlim** [comment]
And we don't  need to update the remote scene tree or remote inspector so frequently.
There are settings to change the interval time for that in the editor settings.

**Comment by reduz** [comment]
the ideal fix would be to send the whole tree over many frames, though not sure if the problem is the scene tree or the property editing.

**Comment by akien-mga** [comment]
Is this still valid in the current master branch?

**Comment by mrcdk** [comment]
Yes, it's still an issue in current master https://github.com/godotengine/godot/commit/8e8240450edc48f4fb379e593a4ef7f54ab656b7

**Comment by akien-mga** [comment]
I can't reproduce on my hardware with the test scene provided, but it's a relatively simple one (might be more visible with a lot more nodes to sync).

At any rate, the debugger defaults to Local now so the problem only triggers when you pick Remote manually, in which case it's a tradeoff for the additional information you requested so it's not too bad.

Could still be improved as mentioned by @reduz, but not critical for 3.0.

**Comment by AWKgamedev** [comment]
I'm still getting stutters with 3.0 rc1, but it's only really noticeable with 1000+ nodes.

**Comment by akien-mga** [comment]
Is this still reproducible in the current master branch? I think it might have been fixed since then.

**Comment by mrcdk** [comment]
It's still happening in https://github.com/godotengine/godot/commit/382f0991ee7ece2a99cfbdbf641d46777b8c8edf

**Comment by Dentoid** [comment]
I don't know if this helps or not, but before I saw this issue I threw a profiler on the problem and this is the result from that:
![image](https://user-images.githubusercontent.com/495668/58353636-4d02a780-7e6f-11e9-8d7b-49b2115a6b93.png)
Feel free to ignore it if it's useless information, but I thought it could be a hint on what part of the scene update is taking time. (This is in an "artifical" scene with just hundreds of duplicated objects.)

**Comment by akien-mga** [comment]
Still relevant in the current master branch (a1c396e55) with the test project in OP.

**Comment by Zireael07** [comment]
Still relevant, especially visible on projects with many nodes.

**Comment by rainlizard** [comment]
Dear god I was about to throw Godot in the garbage bin thinking it couldn't handle a decent amount of nodes without stuttering. It was very frustrating not knowing where the lag was coming from. The cause was this Remote Scenetree, it's almost more devastating to performance than the game itself!

I've got a complex tree, so my mass of nodes is quite buried (the problem isn't me being blind with the scenetree).

Suggestions:

- Put a big warning on the "switch to remote scenetree" option. You can say this whole problem is self-inflicted because I didn't leave the Godot options at their default values, but that's silly, I had no idea why it wasn't enabled, it seemed like a good thing to enable so I enabled it. I fully expect every other person who looks through Godot options will do the same. Then one day they'll be complaining on social media about how Godot is a laggy engine.

- The profiler wasn't showing the source of this stutter. Lag from print() statements seems to be detectable since they're gdscript, but it might be worth considering putting other editor stuff that slows down the game into the profiler. So that we can realize what's going on.

- If nodes are folded in the scenetree, maybe don't stress out the editor updating several thousand hidden things? I don't know how the updating works so this might not be feasible. But if the "obvious issue" is a "huge list of nodes", well it's not obvious, it's a folded issue you can't even see.

- Only update the Remote Scenetree when the editor window is focused, not the game window (this suggestion might be unpopular).

Or maybe a mouse tooltip:
![Untitled](https://user-images.githubusercontent.com/15337628/114375288-db00ec00-9bc7-11eb-806a-26f1019ccc52.png)

**Comment by YuriSizov** [comment]
Removing the high priority label since the assessment for this issue was to postpone the resolution for now.

**Comment by KoBeWi** [comment]
Can anyone still reproduce this issue in Godot 4?

I tested with my super big project™ and it does not occur. I have another project where the issue is more apparent, but it's for 3.x unfortunately. I tried to do some minimal repro and I don't see any stutter with 10k+ nodes with script (although they didn't have any behavior). Would be nice if someone could confirm.

**Comment by DmitriySalnikov** [comment]
v4.0.beta5.official [89a33d28f]

Only 1 basic script for `CharacterBody3D` and many nodes
![image](https://user-images.githubusercontent.com/7782218/202799837-054d7934-74e9-4105-965b-3ed9dcfdcfc6.png)
![image](https://user-images.githubusercontent.com/7782218/202797280-08ad64b3-844f-417c-b7bb-3cfb5c49f049.png)
![Godot_v4 0-beta5_win64_jfxOCAtxx3](https://user-images.githubusercontent.com/7782218/202802051-594fa3d4-1b39-42ff-870a-bb88003045f5.gif)

This scene is imported via gltf from blender using the `-col` and `-colonly` flags. In blender this scene contains 2172 objects.
Extracted from Sonic Adventure DX.
![image](https://user-images.githubusercontent.com/7782218/202798982-d53f60aa-a5de-4042-83be-dcf6f8422b68.png)
